pkg/controller/volume/persistentvolume/index.go

```diff
@@ -157,6 +170,31 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *v1.PersistentVol
 	return nil, nil
 }
 
+// checkVolumeModeMatches is a convenience method that checks volumeMode for PersistentVolume
+// and PersistentVolumeClaims along with making sure that the Alpha feature gate BlockVolume is
+// enabled.
+// This is Alpha and could change in the future.
+func checkVolumeModeMisMatches(pvcSpec *v1.PersistentVolumeClaimSpec, pvSpec *v1.PersistentVolumeSpec) bool {
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		if pvSpec.VolumeMode != nil && pvcSpec.VolumeMode != nil {
+			requestedVolumeMode := *pvcSpec.VolumeMode
+			pvVolumeMode := *pvSpec.VolumeMode
+
+			if requestedVolumeMode != pvVolumeMode {
+				return true
+			}
+			return false
+		} else if pvSpec.VolumeMode == nil && pvcSpec.VolumeMode == nil {
```

Comment:
An error should be returned for the other conditions.  It means API defaulting has gone horribly wrong.